File: kanban-board/src/app/task/task.ts
import { TaskStatus } from './task-status';
export class Task {
    constructor(public id:number,
                public title: string,
                public description: string,
                public status: TaskStatus){}

    static statusFromString(statusStr: string) : TaskStatus{
        switch(statusStr){
            case "New": return TaskStatus.New
            case "Dev": return TaskStatus.Dev
            case "Test": return TaskStatus.Test
            case "Deploy": return TaskStatus.Deploy 
            default: return TaskStatus.New
        };
    }
}

File: kanbanboardbe/src/main/java/com/evgeni/kanbanboardbe/domain/Task.java
package com.evgeni.kanbanboardbe.domain;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

/**
 * Created by evgeni on 9/23/16.
 */
@Data
@Entity
public class Task {
    @Id
    @GeneratedValue
    private long id;
    private String title;
    private String description;
    private TaskStatus status;
}
